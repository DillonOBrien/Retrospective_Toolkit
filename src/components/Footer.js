import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import CssBaseline from '@material-ui/core/CssBaseline';
import useScrollTrigger from '@material-ui/core/useScrollTrigger';
import Slide from '@material-ui/core/Slide';

const useStyles = makeStyles( () => ({

    footer: {
        marginTop: 'calc(5% + 100px)',
        bottom: 0
    }
}));

function HideOnScroll(props) {
    const { children } = props;
    // Note that you normally won't need to set the window ref as useScrollTrigger
    // will default to window.
    // This is only being set here because the demo is in an iframe.
    const trigger = useScrollTrigger({ target: undefined });
  
    return (
      <Slide appear={false} direction="up" in={!trigger}>
        {children}
      </Slide>
    );
  }

  HideOnScroll.propTypes = {
    children: PropTypes.element.isRequired,

  };


const Footer = () => {

    return (

        <React.Fragment>
        <CssBaseline />
        <HideOnScroll {...props}>
          <AppBar>
            <Toolbar>
              <Typography variant="h6">&copy; 2020 - Retrospective Toolkit</Typography>
            </Toolbar>
          </AppBar>
        </HideOnScroll>
        <Toolbar />
      </React.Fragment>

    );


}

export default Footer;