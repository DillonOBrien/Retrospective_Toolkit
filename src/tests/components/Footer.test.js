import React from "react";
import { shallow } from "enzyme";
import Footer from "../../components/Footer";

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Footer />);
});

describe("Components/Footer Testing", () => {
  // Snapshot for header
  test("SNAPSHOT: Footer", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
