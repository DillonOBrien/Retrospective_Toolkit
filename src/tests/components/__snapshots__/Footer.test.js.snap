// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Footer Testing SNAPSHOT: Footer 1`] = `
<Fragment>
  <WithStyles(ForwardRef(AppBar))
    className="makeStyles-footer-2"
  >
    <WithStyles(ForwardRef(Toolbar))
      className="makeStyles-root-1"
    >
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-footer__text-3"
      >
        Â© 2021 - Retrospective Toolkit
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-grow-4"
      />
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-footer__text-3"
      >
        Developed by High Rollers
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</Fragment>
`;
